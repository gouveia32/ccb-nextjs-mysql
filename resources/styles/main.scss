
// Main
@import "./base/base";
@import "./base/variables";
@import "./base/scrollbar";
