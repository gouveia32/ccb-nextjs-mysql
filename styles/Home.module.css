@media only screen and (min-width: 764px) {
  .homeLanding {
    flex-direction: row;
    background-color: red;
    justify-content: space-between;
  }
}

.homeLanding {
  display: flex;
  flex-direction: column;
  justify-content: center;
}
