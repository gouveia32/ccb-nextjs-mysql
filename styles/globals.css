html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen,
    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

#page-wrap {
  padding-bottom: 0px;
  padding-top: 0px;
}

@media only screen and (min-width: 764px) {
  .bm-burger-button {
    display: none;
  }
}

@media only screen and (max-width: 763px) {
  .nv-links {
    display: none;
  }
  .nv-link {
    display: none;
  }
  .nav-link-active {
    display: none;
  }
}

/* Contact icons */

.contact-ways-icon {
  transition: all 0.8s;
}

.contact-ways-icon:hover {
  color: #ffd700;
  transform: scale(1.2);
}

/* scroll bar of pricing table  */
.table-scroll-bar::-webkit-scrollbar {
  height: 8px;
  background-color: #f5f5f5;
}

.table-scroll-bar::-webkit-scrollbar-track {
  box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
  background-color: #f5f5f5;
}

.table-scroll-bar::-webkit-scrollbar-thumb {
  border-radius: 10px;
  box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
  background-color: #ffd700;
}

/*facebook icon */
.fb-icon {
  transition: all 0.8s;
}
.fb-icon:hover {
  box-shadow: 5px 9px 50px -23px #3b5998;
  transform: scale(1.2);
  color: #3b5998;
}

/*twitter icon */
.twitt-icon {
  transition: all 0.8s;
}
.twitt-icon:hover {
  box-shadow: 5px 9px 50px -23px #00acee;
  transform: scale(1.2);
  color: #00acee;
}

/*youtube icon */
.yt-icon {
  transition: all 0.8s;
}

.yt-icon:hover {
  box-shadow: 5px 9px 50px -23px #c4302b;
  transform: scale(1.2);
  color: #c4302b;
}

/*instagram icon */
.inst-icon {
  transition: all 0.8s;
}

.inst-icon:hover {
  box-shadow: 5px 9px 50px -23px white;
  transform: scale(1.2);
  color: white;
}

/* footer */
.footer-link {
  color: rgb(112, 107, 107);
}

.footer-link:hover {
  color: #e4d687;
}

/* nav links*/

.nav-link-active-side-bar {
  color: #ffd700;
  font-weight: bold;
  font-size: larger;
  margin-bottom: 15px;
}

.nav-link-active {
  color: #ffd700 !important;
  font-weight: bold;
  font-size: larger;
  margin-right: 10px;
}

.nv-links {
  display: flex;
  justify-content: center;
  align-items: center;
}

.nv-link {
  font-size: larger;
  margin-right: 20px;
  color: #373a47;
  font-weight: bolder;
  transition: all 0.5s;
}
.nv-link:hover {
  color: #ffd700;
}

/* Position and sizing of burger button */
.bm-burger-button {
  position: fixed;
  width: 50px;
  height: 50px;
  right: 15px;
  top: 7px;
}

/* Color/shape of burger icon bars */
.bm-burger-bars {
  background: #373a47;
}

/* Color/shape of burger icon bars on hover*/
.bm-burger-bars-hover {
  background: #a90000;
}

/* Position and sizing of clickable cross button */
.bm-cross-button {
  height: 24px;
  width: 24px;
}
/* Color/shape of close button cross */

.bm-cross {
  background: transparent;
}

/*
Sidebar wrapper styles
Note: Beware of modifying this element as it can break the animations - you should not need to touch it in most cases
*/
.bm-menu-wrap {
  position: fixed;
  height: 100%;
}

/* General sidebar styles */
.bm-menu {
  background: white;
  padding: 2.5em 1.5em 0;
  font-size: 1.15em;
  border-left: 1.5px solid #ff0000;
}

/* Morph shape necessary with bubble or elastic */
.bm-morph-shape {
  fill: #373a47;
}

/* Wrapper for item list */
.bm-item-list {
  font-weight: bolder;
  padding: 0.8em;
  display: flex;
  flex-direction: column;
  height: 374px;
  transition: all 0.5s;
}

/* Individual item */
.bm-item {
  display: block;
}

/* Styling of overlay */
.bm-overlay {
  background: rgba(0, 0, 0, 0.3);
}

/* Individual item */
.bm-item {
  display: inline-block;
  color: #373a47;
  margin-bottom: 10px;
  text-align: left;
  text-decoration: none;
  transition: color 0.2s;
}

.bm-item:hover {
  color: #ffd700;
}
