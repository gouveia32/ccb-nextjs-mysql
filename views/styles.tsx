import styled from "styled-components";
import { motion } from "framer-motion";
import { device } from "../resources/styles/utils/media-query-utils";

const dFlexCentered = `display: flex;
  justify-content: center;
  align-items: center;`;

export const LandingPageWrapper = styled(motion.section)`
  height: 100%;
  flex-direction: column;
  ${dFlexCentered}

  @media only screen and ${device.mobileL} {
    height: max-content;
    margin-left: -50px;
    margin-top: 2rem;
  }
`;

export const LandingPageContentWrapper = styled(motion.div)`
  ${dFlexCentered}
  
  @media only screen and ${device.mobileL} {
    flex-direction: column;
  }
`;

export const LandingPageCardWrapper = styled(motion.div)`
  flex-direction: column;
  ${dFlexCentered}

  @media only screen and ${device.mobileL} {
    margin: 0 !important;
  }
`;

export const LandingPageHeading = styled.h1`
  margin-bottom: 1rem;
  font-weight: bold;
  font-size: 4rem;
  color: #f5b500;
`;

export const LandingPageSubHeading = styled.h2`
  margin-top: 2rem;
  margin-bottom: 1rem;
  color: #8a8a8a;
`;

interface LandingPageBannerInterface {
  imageUrl?: string;
  color?: string;
  position?: string;
}

export const LandingPageBannerWrapper = styled.div`
  position: relative;
  height: 300px;
  width: 300px;
  overflow: hidden;
`;

export const LandingPageBanner = styled(motion.div)<LandingPageBannerInterface>`
  height: 300px;
  width: 300px;
  transform-origin: top;
  font-size: 2.8rem;
  font-family: "Roboto Slab", serif;
  font-weight: lighter;
  padding: 1rem;
  top: 0;
  background-size: cover;
  overflow: hidden;
  display: flex;
  flex-direction: column;

  position: ${(props) => (props.position ? props.position : "relative")};
  background: ${(props) =>
    props.imageUrl
      ? `url(${props.imageUrl})`
      : props.color
      ? props.color
      : "#ffe900"};
`;

export const LandingPageBannerText = styled(motion.h1)`
  font-family: "Shadows Into Light Two", cursive;
  font-size: 1.8rem;
  left: 0;
  padding: 10px;
  position: absolute;
  right: 0;
  text-align: center;
  top: 50%;
  -webkit-transform: translateY(-50%) rotate(-4deg);
  transform: translateY(-50%) rotate(-4deg);
`;
