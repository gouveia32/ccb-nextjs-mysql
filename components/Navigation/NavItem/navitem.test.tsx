import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import NavigationItem, { NavigationItemProps } from "./navitem.component";
import { NavItem, NavItemContent, NavItemIcon } from "./navitem.styles";

const useRouter = jest.spyOn(require("next/router"), "useRouter");

describe("NavItem component", () => {
  let wrapper: ShallowWrapper;
  let mockOnClick: any;

  beforeEach(() => {
    mockOnClick = jest.fn();

    useRouter.mockImplementationOnce(() => ({
      push: () => null,
    }));

    const mockProps: NavigationItemProps = {
      icon: null,
      isActive: false,
      isOpen: false,
      isTag: false,
      name: "",
      onClick: mockOnClick,
      url: "",
    };

    wrapper = shallow(<NavigationItem {...mockProps} />);
  });

  it("should render NavItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onClick when NavItem si clicked", () => {
    wrapper.find(NavItem).simulate("click");
    expect(mockOnClick).toBeCalled();
  });

  it("should render NavItemIcon, NavItemContent", () => {
    expect(wrapper.find(NavItemIcon)).toHaveLength(1);
    expect(wrapper.find(NavItemContent)).toHaveLength(1);
  });
});
