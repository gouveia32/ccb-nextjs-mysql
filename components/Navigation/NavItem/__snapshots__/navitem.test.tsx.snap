// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem component should render NavItem component 1`] = `ShallowWrapper {}`;
